### PR TITLE
Use Axes instead of axes in galleries

### PR DESCRIPTION
**Pull request by timhoffm**
Part of #18726.

On a side-note: What I have learned from this: Even when knowing the definitions and differences between `Axes` and *axes* as a simple plural of axis, it is quite challenging to sort out which one to use in a couple of places. I have now a vague impression how confusing that naming may be to our users.

### CODE REVIEW

**Comment by timhoffm** [comment]
> If we still do this anywhere, we should probably change it to x and y (and/or whatever else) axis to save everyone some trouble.

Possibly in some places, though that's not practical everywhere. There are edge cases like *axes coordinates* that I have not touched and others cases where additional qualification is not needed or rather distracting when the context is clear. One could grep through the remaining "axes", but overall I think that's a second-order improvement. Consistently using `Axes` where it clearly denotes the object and is not a multiple of axis is the main improvement to reduce confusion.

File: galleries/examples/spines/spines_dropped.py [hunk 1]
```diff
@@ -3,7 +3,7 @@
 Dropped spines
 ==============
 
-Demo of spines offset from the axes (a.k.a. "dropped spines").
+Demo of spines offset from the Axes (a.k.a. "dropped spines").
```
Comment (ksunden) [inline galleries/examples/spines/spines_dropped.py:6]: This one I'm not quite sure we want to evoke the `Axes` object here, as the spines are _part_ of the `Axes` object, and they are not offset from such. I could go either way, but feels slightly different.

File: galleries/tutorials/pyplot.py [hunk 2]
```diff
@@ -26,7 +26,7 @@
 # In :mod:`matplotlib.pyplot` various states are preserved
 # across function calls, so that it keeps track of things like
 # the current figure and plotting area, and the plotting
-# functions are directed to the current axes (please note that "axes" here
+# functions are directed to the current Axes (please note that "axes" here
 # and in most places in the documentation refers to the *axes*
 # :ref:`part of a figure <figure_parts>`
 # and not the strict mathematical term for more than one axis).
```
Comment (ksunden) [inline galleries/tutorials/pyplot.py:32]: This paragraph is slightly meta here, and seems like we may want to stay capalized for  more of this sentence, especially the quoted instance.

File: lib/matplotlib/figure.py
Comment (ksunden) [inline lib/matplotlib/figure.py:1]: this file is not in the galleries, fyi

**Review by ksunden — COMMENTED** [review]
Similar to #27105, a couple notes here and there, but largely agree.

I so still have the `figure or Axes` feels incongruent when in close proximity and exactly one of them is capitalized, though that seemed less common as I read through this set of files.

File: galleries/tutorials/pyplot.py (hunk 2 above)
Comment (timhoffm) [reply galleries/tutorials/pyplot.py]: I've reworded the paragraph.

File: lib/matplotlib/figure.py
Comment (timhoffm) [reply lib/matplotlib/figure.py:1]: removed.

**Review by ksunden — APPROVED** [review]

**Pull request merged**